### PR TITLE
fix/remove file id

### PR DESCRIPTION
**Pull request by jamestjw**
- Change file_id to saved_filename
- Increase length of unique file ID

This affects all file downloads, updating and deleting. Highly recommend testing this out, I did it once but I might have missed something.

### CODE REVIEW

File: public/create_marked_entity.php [hunk 1]
```diff
@@ -33,16 +33,15 @@
     // TODO: Make this more convenient
     $target_dir = "uploads/";
     // TODO: Improve the file ID
-    $file_id = uniqid().basename($_FILES["marked_entity_file"]["name"]);
-    $target_file = $target_dir . $file_id;
+    $saved_filename = uniqid("", true);
```
Comment (jamestjw) [inline public/create_marked_entity.php:36]: "" prefix, second arg makes the unique ID longer

File: public/create_marked_entity.php [hunk 2]
```diff
@@ -33,7 +33,7 @@
     // TODO: Make this more convenient
     $target_dir = "uploads/";
     // TODO: Improve the file ID
-    $file_id = uniqid().basename($_FILES["marked_entity_file"]["name"]);
+    $file_id = uniqid("", true);
```
Comment (jamestjw) [inline public/create_marked_entity.php:36]: this gives a longer ID to avoid collisions

**Review by drouba — APPROVED** [review]
just make sure the delete function works as intended before merging please

File: public/marked_entities/delete_student_file.php [hunk 3]
```diff
@@ -36,7 +36,7 @@
 $marked_entity_file_change->file_name = $file->file_filename;
 $marked_entity_file_change->set_action("delete");
 
-$file->delete();
+$file->delete("file_id");
```
[the remark this answers is not in the document]
Comment (jamestjw) [reply public/marked_entities/delete_student_file.php:39]: yep, seems to work

**Pull request merged**